CMakeLists.txt

```diff
@@ -3,14 +3,31 @@ project(exit2destination LANGUAGES C CXX)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -std=c++11")
 
-find_package(Boost REQUIRED)
+#find_package(Boost REQUIRED)
 find_package(Threads REQUIRED)
 
 # Vendored Libosmium
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/cmake")
-set(OSMIUM_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include")
-find_package(Osmium REQUIRED COMPONENTS pbf)
+if (ENABLE_MASON)
+    set(MASON_COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/.mason/mason)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/.mason/mason.cmake)
+
+    mason_use(libosmium VERSION 2.10.3 HEADER_ONLY)
+    include_directories(${MASON_PACKAGE_libosmium_INCLUDE_DIRS})
+    set(OSMIUM_INCLUDE_DIRS ${MASON_PACKAGE_libosmium_INCLUDE_DIRS})
+
+    mason_use(protozero VERSION 1.4.2 HEADER_ONLY)
+    include_directories(${MASON_PACKAGE_protozero_INCLUDE_DIRS})
+    set(PROTOZERO_INCLUDE_DIRS ${MASON_PACKAGE_protozero_INCLUDE_DIRS})
+else()
+    # expects a vendored libosmium at /third_party
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/cmake")
+    set(OSMIUM_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include")
+    find_package(Osmium)
+endif()
+
+find_package(ZLIB REQUIRED)
+include_directories(${ZLIB_INCLUDE_DIRS})
 
 add_executable(${PROJECT_NAME} ${PROJECT_NAME}.cc)
-target_link_libraries(${PROJECT_NAME} ${OSMIUM_LIBRARIES} ${Boost_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-target_include_directories(${PROJECT_NAME} SYSTEM PRIVATE ${OSMIUM_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
+target_link_libraries(${PROJECT_NAME} ${OSMIUM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${ZLIB_LIBRARY})
```

Comment:
Per above this is correct and needed since `rewrite-exit-destination-signage` depends on zlib to link correctly. Per chat with @karenzshea for the `find_package(Osmium)` case it is not needed so a good solution would be moving the blocks to be different for the mason approach vs `find_package(Osmium)` approach. The two can live together but have different needs.